sentry_sdk/metrics.py

```diff
@@ -609,7 +609,7 @@ def record_code_location(
                 )
 
     @metrics_noop
-    def need_code_loation(
+    def need_code_location(
```

Comment:
I would advocate making a deprecated `need_code_loation` that calls `need_code_location` for backwards compatibility to make this change easier for users, unless there is a reason why this would not be desirable.

And, are we using this method anywhere in the code where it needs to be updated?

Reply:
I think this would be overkill. 

I do not get where this function is used at all. It is not called in the SDK, and also not in `sentry` or `getsentry`. So I guess just renaming it is fine.

Reply:
I think this would be overkill.

I do not get where this function is used at all. It is not called in the SDK, and also not in sentry or getsentry. So I guess just renaming it is fine.

Reply:
Agree with Anton, I don't imagine anyone's actually using this function. I think we can afford to rename this in the major as long as we call it out properly in the MIGRATION_GUIDE.

Reply:
Fair enough, if few people use it then I agree